tests/zfs-tests/include/libtest.shlib

```diff
@@ -1595,6 +1591,44 @@ function destroy_pool #pool
 	return 0
 }
 
+# Return 0 if destroy successfully or the dataset exists; $? otherwise
+# Note: In local zones, this function should return 0 silently.
+#
+# $1 - dataset name
+# $2 - custom arguments for zfs destroy
+# Destroy dataset with the given parameters.
+
+function destroy_dataset #dataset #args
+{
+	typeset dataset=$1
+	typeset mtpt
+	typeset args="-f"
```

Comment:
We probably shouldn't default to `-f`, instead how about `typeset args=${2:-""}`.  Then you can drop the `args` assignment below, and callers won't need to pass a second argument unless requesting thing the `-f` behavior.